Assignment3/Q1.java

```diff
@@ -0,0 +1,148 @@
+import java.util.*;
+
+class Q1
+{
+	public static void main(String args[])
+	{
+		char[][] grid = new char[2][3];
+		grid[0][0] = 'A';
+		grid[0][1] = 'A';
+		grid[0][2] = 'R';
+		grid[1][0] = 'T';
+		grid[1][1] = 'C';
+		grid[1][2] = 'D';
+
+		TreeSet<String> answer = findWords(2,3,grid,new Dictionary());
+
+		Iterator it = answer.iterator();
+
+		while(it.hasNext())
+		{
+			System.out.println(it.next());
+		}
+	}
+
+
+
+	public static TreeSet<String> findWords (int rows, int columns, char[][] grid, Dictionary dictionary)
+	{
```

Comment:
A couple of questions:
1) Why pass through the rows and columns? The grid knows its size.
2) If you do pass through the rows and columns, it would be good to do some data validation. Check each row/column of the grid is as big as it should be.